### PR TITLE
proxy: Migrate jwt module away from anyhow

### PR DESCRIPTION
**Pull request by cloneable**

### CODE REVIEW

File: proxy/src/auth/backend/jwt.rs [hunk 1]
```diff
@@ -561,6 +555,68 @@ impl Drop for JwkRenewalPermit<'_> {
...
+    JwtEncoding(#[from] JwtEncodingError),
+
+    #[error("invalid JWT token audience")]
+    InvalidJwtTokenAudience,
+
+    #[error("JWT token has expired")]
+    JwtTokenHasExpired,
+
+    #[error("JWT token is not yet ready to use")]
+    JwtTokenNoYettReadyToUse,
```
Comment (conradludgate) [inline proxy/src/auth/backend/jwt.rs:577]: nit: typo.

**Review by conradludgate — APPROVED** [review]
looks good. thanks

File: proxy/src/auth/backend/jwt.rs (hunk 1 above)
Comment (cloneable) [reply proxy/src/auth/backend/jwt.rs]: Done.

**Comment by cloneable** [comment]
I now also removed anyhow from the FetchAuthRules paths. Ptal

**Review by conradludgate — APPROVED** [review]

**Pull request merged**